src/Rules/DateBetween.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Spatie\ValidationRules\Rules;
+
+use Carbon\Carbon;
+use Illuminate\Contracts\Validation\Rule;
+
+class DateBetween implements Rule
+{
+    /** @var \Carbon\Carbon */
+    protected $start;
+
+    /** @var \Carbon\Carbon */
+    protected $end;
+
+    /** @var bool */
+    protected $orEquals = false;
+
+    public function __construct(Carbon $start, Carbon $end)
+    {
+        $this->start = $start->copy();
+
+        $this->end = $end->copy();
+    }
+
+    public function orEquals(bool $orEquals = true): DateBetween
+    {
+        $this->orEquals = $orEquals;
+
+        return $this;
+    }
+
+    public function withoutTime(): DateBetween
+    {
+        $this->start->startOfDay();
+
+        $this->end->endOfDay();
+
+        return $this;
+    }
+
+    public function passes($attribute, $value)
+    {
+        $date = Carbon::make($value);
```

Comment:
If a format is given via the constructor, let's use it here. Also add a few tests to make sure that date format is being respected.

Reply:
Correct me if I'm wrong, but doesn't the `date_format` rule already does this? What's the added value of adding it also in this rule?

Reply:
Even with that rule in place how would `Carbon::make($value)` know in which date format the `$value` is?

Reply:
`Carbon::make` will figure out the format itself: https://github.com/briannesbitt/Carbon/blob/version-1.34/src/Carbon/Carbon.php#L972

That's why I wanted to use `make`. Combined with the `date_format` rule, you'll always get a valid date.

Reply:
Assuming we have a `date_format` rule for an attribute, the `Carbon::make` call here cannot know which format was used by that rule. What will `Carbon::make` produce when given 01-03-2018. 1st of March? Or 3th of January?

Reply:
You're right. I mistakenly thought that `date_format` would cast the value to a Carbon object before passing it to the next rule. I've added format support.